packages/core/src/internal/proc.js

```diff
@@ -294,8 +286,8 @@ export default function proc(
           This will jump to the finally block
         **/
         result = is.func(iterator.return) ? iterator.return(TASK_CANCEL) : { done: true, value: TASK_CANCEL }
-      } else if (arg === CHANNEL_END) {
-        // We get CHANNEL_END by taking from a channel that ended using `take` (and not `takem` used to trap End of channels)
+      } else if (isTerminated(arg)) {
```

Comment:
wouldnt `shouldTerminate` be more appropriate? just thinking out loud, let me know what do u think

Reply:
well, actually we just received request to terminate, so probably you are right and it should be called `should...`